src/templates/kit/screens/FundraisingScreen.js

```diff
@@ -0,0 +1,573 @@
+    }),
+    {}
+  )
+}
+
+function useConfigureFields(data) {
+  const [fields, _update] = useReducer(reduceFields, dataDefaults(data))
+
+  // Update a field by using its name and value as parameters
+  const update = useCallback((...params) => _update(params), [])
+
+  // Bind the update function to a callback passing the value, e.g.:
+  //
+  //  <PercentageField
+  //   onChange={bindUpdate('fieldName')}
+  //   value={fieldValue}
+  //  />
+  //
+  const bindUpdate = useCallback(name => value => update(name, value), [update])
+
+  return { fields, update, bindUpdate }
+}
+
+function updateVestingSchedule(fields, value) {
+  const vestingSchedule = Math.max(-1, intDef(value, -1))
+  const cliffPeriod = Math.max(
+    -1,
+    Math.min(fields.cliffPeriod, vestingSchedule - 1)
+  )
+  return { ...fields, vestingSchedule, cliffPeriod }
+}
+
+function updateCliffPeriod(fields, value) {
+  const cliffPeriod = Math.max(-1, intDef(value, -1))
+  const vestingSchedule = Math.max(-1, fields.vestingSchedule, cliffPeriod + 1)
+  return { ...fields, vestingSchedule, cliffPeriod }
+}
+
+function reduceFields(fields, [field, value]) {
+  const updateField = value => ({ ...fields, [field]: value })
+
+  if (field === 'description') {
+    return updateField(value)
+  }
+
+  if (
+    field === 'monthlyAllowance' ||
+    field === 'upfrontCosts' ||
+    field === 'targetGoal' ||
+    field === 'fundingPeriod'
+  ) {
+    return updateField(intDef(value))
+  }
+
+  if (field === 'tokensOffered') {
+    return updateField(value)
+  }
+
+  if (field === 'initialPricePerToken') {
+    return updateField(intDef(value))
+  }
+
+  if (field === 'expectedGrowth') {
+    return updateField(intDef(value))
+  }
+
+  if (field === 'vestingSchedule') {
+    return updateVestingSchedule(fields, value)
+  }
+
+  if (field === 'cliffPeriod') {
+    return updateCliffPeriod(fields, value)
+  }
+
+  if (field === 'batchLength') {
+    return updateField(intDef(value))
+  }
+
+  if (field === 'slippageDai') {
+    return updateField(intDef(value))
+  }
+
+  if (field === 'slippageAnt') {
+    return updateField(intDef(value))
+  }
+
+  if (field === 'initialMonthlyAllocationDai') {
+    return updateField(intDef(value))
+  }
+
+  if (field === 'initialMonthlyAllocationAnt') {
+    return updateField(intDef(value))
+  }
+
+  if (field === 'initialTapFloorsDai') {
+    return updateField(intDef(value))
+  }
+
+  if (field === 'initialTapFloorsAnt') {
+    return updateField(intDef(value))
+  }
+
+  if (field === 'maximumMonthlyAllocationAndFloorIncreases') {
+    return updateField(intDef(value))
+  }
+
+  return fields
+}
+
+function FundraisingScreen({
+  dataKey,
+  screenProps: { back, data, next, screenIndex, screens },
+}) {
+  const screenData = (dataKey ? data[dataKey] : data) || {}
+
+  const theme = useTheme()
+  const [tab, setTab] = useState(0)
+  const { fields, bindUpdate } = useConfigureFields(
+    screenData.fundraising || {}
```

Comment:
This `|| {}` is not needed no? You are already making sure `screenData` is an object above.